composeApp/src/commonMain/kotlin/org/ooni/probe/ui/settings/SettingsScreen.kt

```diff
@@ -1,30 +1,604 @@
+import ooniprobe.composeapp.generated.resources.category_porn
+import ooniprobe.composeapp.generated.resources.category_prov
+import ooniprobe.composeapp.generated.resources.category_pubh
+import ooniprobe.composeapp.generated.resources.category_rel
+import ooniprobe.composeapp.generated.resources.category_srch
+import ooniprobe.composeapp.generated.resources.category_xed
+import ooniprobe.composeapp.generated.resources.ic_settings
+import ooniprobe.composeapp.generated.resources.notifications
+import ooniprobe.composeapp.generated.resources.outline_info
+import ooniprobe.composeapp.generated.resources.privacy
+import ooniprobe.composeapp.generated.resources.proxy
+import ooniprobe.composeapp.generated.resources.send_email
 import ooniprobe.composeapp.generated.resources.settings
+import org.jetbrains.compose.resources.DrawableResource
+import org.jetbrains.compose.resources.StringResource
+import org.jetbrains.compose.resources.painterResource
 import org.jetbrains.compose.resources.stringResource
-import org.jetbrains.compose.ui.tooling.preview.Preview
-import org.ooni.probe.ui.theme.AppTheme
+import org.ooni.probe.data.repositories.PreferenceCategoryKey
+import org.ooni.probe.data.repositories.SettingsKey
+import org.ooni.probe.ui.settings.category.SettingsDescription
 
 @Composable
-fun SettingsScreen() {
+fun SettingsScreen(onNavigateToSettingsCategory: (SettingsViewModel.Event) -> Unit) {
     Column {
         TopAppBar(
             title = {
                 Text(stringResource(Res.string.settings))
             },
         )
+
+        SettingsCategoryItem.getSettingsItems().forEach { item ->
+            SettingsItemView(
+                icon = item.icon,
+                title = item.title,
+                modifier =
+                    Modifier.clickable {
+                        onNavigateToSettingsCategory(
+                            item.routeToSettingsCategory(),
+                        )
+                    },
+            )
+        }
     }
 }
 
-@Preview
 @Composable
-fun SettingsScreenPreview() {
-    AppTheme {
-        SettingsScreen()
+fun SettingsItemView(
+    icon: DrawableResource?,
+    title: StringResource,
+    modifier: Modifier,
+) {
+    ListItem(
+        leadingContent = {
+            icon?.let {
+                Image(
+                    modifier = Modifier.height(24.dp).width(24.dp),
+                    painter = painterResource(it),
+                    contentDescription = stringResource(title),
+                )
+            }
+        },
+        headlineContent = { Text(stringResource(title)) },
+        modifier = modifier,
+    )
+}
+
+open class PreferenceItem(
+    open val title: StringResource,
+    open val icon: DrawableResource? = null,
+    open val type: PreferenceItemType,
+    open val key: SettingsKey,
+    open val supportingContent:
+        @Composable()
+        (() -> Unit)? = null,
+)
+
+data class SettingsItem(
```

Comment:
I feel this classes should be in separate files, especially because they are not just related with this screen.